### PR TITLE
DAOS-2540 VOS: add new API evt_drain

### PR DESCRIPTION
**Pull request by gnailzenh**
This patch includes a few EVtree API changes:

- Add a new API evt_drain
  This function drains data extents from the tree, each time it frees
  an extent it consumes a "credit" which is input paramter of this
  function. It returns if all input credits are consumed, or the tree
  is empty, in the later case, it also destroys the evtree.

- Add a new function table for evt_desc
  It is passed in while openning an evtree, this function table defines
  callbacks for:
  . add evt_desc to undo log
  . remove evt_desc to undo log
  . check log status of evt_desc
  . free storage space for evt_desc

Signed-off-by: Liang Zhen <liang.zhen@intel.com>

### CODE REVIEW

File: src/include/daos_srv/evtree.h [hunk 1]
```diff
@@ -61,6 +61,48 @@ struct evt_desc {
...
+	/**
+	 * Availability check, it is for data tracked by DTX undo log.
+	 * It is optional, EVTree always treats data extent is available if
+	 * this method is absent.
+	 */
+	int		(*dc_log_status_cb)(struct umem_instance *umm,
+					    struct evt_desc *desc,
+					    int intent, void *args);
+	void		 *dc_log_status_args;
+	/** Add a descriptor to undo log */
```
Comment (jolivier23) [inline src/include/daos_srv/evtree.h]: this is an instance where if we had no direct pointers, a callback would be unecessary

**Comment by gnailzenh** [comment]
I'm going to refresh this patch because it got "timeout" without any other hint, let me try again and see if it's reproducible or not
